src/rgw/rgw_sal_motr.cc

```diff
@@ -226,6 +226,9 @@ int MotrUser::list_buckets(const DoutPrefixProvider *dpp, const string& marker,
   if (rc < 0) {
     ldpp_dout(dpp, 0) <<  __func__ << ": ERROR: NEXT query failed. " << rc << dendl;
     return rc;
+  } else if (rc == 0) {
+    ldpp_dout(dpp, 0) <<  __func__ << ": No buckets to list. " << rc << dendl;
+    return rc;
```

Comment:
each instance of `else if (rc == 0)` from this PR can be shifted to the `if` condition. `if (rc <= 0)`
**if we want to avoid `else if`,**
can anyone suggest a good error msg in that case that will cover both error and zero entries case?

Reply:
`No buckets` is not an error case so no need to put them together. 
The behavoir will remain same if you return from here, right? `aws s3 ls` will still return nothing?